core/src/graphman/core/create.rs

```diff
@@ -0,0 +1,21 @@
+use std::sync::Arc;
+
+use graph::prelude::{anyhow, Error, SubgraphName, SubgraphStore as _};
+use graph_store_postgres::SubgraphStore;
+
+pub fn run(
+    store: Arc<SubgraphStore>,
+    name: String,
+    enable_logging: bool,
+) -> Result<SubgraphName, Error> {
+    let name = SubgraphName::new(name.clone())
+        .map_err(|()| anyhow!("illegal subgraph name `{}`", name))?;
+
+    if enable_logging {
+        println!("creating subgraph {}", name);
+    }
```

Comment:
A better way to print conditionally would be to pass an argument `out: &dyn mut std::io::Write` to these methods and change all instances of `println!(..)` to `writeln!(out, ..)`

From the CLI, we'd pass in `std::io::stdout().lock()`, and from the API, we'd pass in a dummy implementation of `std::io::Write` that does nothing.

Reply:
That's a pretty neat! I have implemented it, Thanks!